core/geotime/src/main/java/mil/nga/giat/geowave/core/geotime/index/dimension/TemporalBinningStrategy.java

```diff
@@ -223,6 +260,20 @@ private Calendar getStartEpoch(
 		final String str = StringUtils.stringFromBinary(binId);
 		final Calendar cal = Calendar.getInstance(TimeZone.getTimeZone(timezone));
 		switch (unit) {
+			case MINUTE:
+				final int minute = Integer.parseInt(str.substring(
+						14,
+						16));
+				cal.set(
+						Calendar.MINUTE,
+						minute);
+			case HOUR:
+				final int hour = Integer.parseInt(str.substring(
+						11,
+						13));
+				cal.set(
+						Calendar.HOUR_OF_DAY,
+						hour);
```

Comment:
it seems that decade is essentially accounted for entirely within getBinSizeMillis() but I kinda agree with @rwgdrummer - a check for decade here would essentially ensure that the year is a multiple of 10 (rounding down if it is now).  Perhaps redundant but it is definitely consistent with what one would expect from getStartEpoch()